flink/src/main/java/org/apache/iceberg/flink/source/IteratorReader.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source;
+
+import java.io.Serializable;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.io.CloseableIterator;
+
+/**
+ * Read a {@link FileScanTask} into a {@link CloseableIterator}
+ */
+public interface IteratorReader<T> extends Serializable {
```

Comment:
The name `IteratorReader` doesn't make sense to me because iterators aren't generally read. They are iterated through. It also seems strange to me to have a method that is essentially a way to create an iterator but is not a `CloseableIterable`. Is there a way to restructure this so that this is a `CloseableIterable` instead?

Last, why does this need to be `Serializable`?

Reply:
what about `FileReader` as class name and change the method name from `open` to `read`? Its purpose is to read a `FileScanTask` into a `CloseableIterator` or `CloseableIterable`.

It needs to be `Serializable` because it gets shipped/serialized from jobmanager to taksmanager during deployment.

Reply:
renamed it to `FileReader`